### DISCUSSION TITLE
Breaking points and width for responsive images?

### DISCUSSION THREAD

**Discussion by SamuelMiller** [twbs/bootstrap#36422] [Q&A]
Hi Folks, I am curious to hear from others what image width sizes they choose for full-width picture/srcset image replacement.  Do you follow Bootstrap's breakpoints? For example, 576w for <576px, 768w for ≥576px, 992w for ≥768px, 1200w for ≥992px, 1400w for ≥1200px, and 2048 for ≥1400px.

As in:
```
<picture>
<source type="image/avif" sizes=”100vw” srcset="/img/lion-sm.avif 576w, /img/lion-md.avif 768w, /img/lion-lg.avif 992w, /img/lion-xlg.avif 1200w" />
<source type="image/webp" sizes=”100vw” srcset="/img/lion-sm.webp 576w, /img/lion-md.webp 768w, /img/lion-lg.webp 992w, /img/lion-xlg.webp 1200w" />
<img sizes="100vw" src="/img/lion.jpeg" srcset="/img/lion-sm.jpeg 576w, /img/lion-md.jpeg 768w, /img/lion-lg.jpeg 992w, /img/lion-xlg.jpeg 1200w" alt="lion">
</picture>
```

**Comment by gerardnico** [comment]
Until I dig a little bit more on [container query](https://css.oddbird.net/rwd/query/explainer/) proposal, I use this [breakpoints](https://github.com/ComboStrap/combo/blob/main/ComboStrap/RasterImageLink.php#L44)

```php
const BREAKPOINTS =
        array(
            "xs" => 375,
            "sm" => 576,
            "md" => 768,
            "lg" => 992
        );
```

**Reply by SamuelMiller** [reply]
Your largest breakpoint is 992. Is that the same for the width of your largest image? Laptops typically have a screen width larger than 1200. Serving a  992 px wide image at 100% width on a larger screen would render a  less than clear image, no?

**Reply by gerardnico** [reply]
Above lg, you get the raw image.
The image width is constrained by its container.

**Comment by SamuelMiller** [comment]
Based on the most common resolutions listed at browserstack.com (see link below), I determined the closest Bootstrap equivalent for both portrait and landscape widths.

```
1920×1080 (8.89%) Bootstrap: 2048w, 1400w (could use 1200w)
1536×864 (3.88%) Bootstrap: 2048w, 992w
1366×768 (8.44%) Bootstrap: 1400w, 992w
414×896 (4.58%) Bootstrap: 576w, 992w
375×667 (3.75%) Bootstrap: 576w, 768w
360×640 (7.28%) Bootstrap: 576w, 768w
```
For now, until I hear something better from someone else, I am going with the following image widths/breakpoints for scrset for full-screen display: 576w, 992w, 1400w and 2048w. ( I didn't include 768w and 1200w as I don't want too many images to deal with.)

For example,

```
<picture>
<source type="image/jpeg" media="(max-width: 575px)" srcset="assets/img/faith-in-action/faith-in-action-cropped-sm.jpg">
<source type="image/jpeg" media="(min-width: 576px)"
srcset="assets/img/faith-in-action/faith-in-action-md.jpg 992w,assets/img/faith-in-action/faith-in-action-xl.jpg 1400w, assets/img/faith-in-action/faith-in-action-xxl.jpg 2048w">
<img src="assets/img/faith-in-action/faith-in-action-cropped-sm.jpg" alt="drummers">
</picture>
```
https://www.browserstack.com/guide/ideal-screen-sizes-for-responsive-design#:~:text=Screen%20Resolution%20Stats%20Worldwide%3A%20Jan%202020%20%E2%80%93%20Jan%202021,-Image%20Source&text=According%20to%20the%20Worldwide%20Screen,360%C3%97640%20(7.28%25)
